### PR TITLE
Add support for mapping messages to CloudEvents

### PR DESCRIPTION
**Pull request by sophokles73**
Added support for mapping UMessages to/from CloudEvents using Protobuf format.

Addresses #203

### CODE REVIEW

File: proto/io/cloudevents/v1/cloudevents.proto
Comment (AnotherDaniel) [inline proto/io/cloudevents/v1/cloudevents.proto:1]: Do we really want to pull this in as a copy? Or maybe revive the download&compile functionality we originally used in build.rs for external proto files?

File: src/cloudevents.rs [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+    UMessageType, UPayloadFormat, UPriority, UUri, UUID,
+};
+use bytes::Bytes;
+use protobuf::{well_known_types::any::Any, Enum, EnumOrUnknown, MessageField};
+
+pub use cloudevents::{cloud_event::CloudEventAttributeValue, CloudEvent};
+
+include!(concat!(env!("OUT_DIR"), "/cloudevents/mod.rs"));
+
+pub const CONTENT_TYPE_CLOUDEVENTS_PROTOBUF: &str = "application/cloudevents+protobuf";
```
Comment (AnotherDaniel) [inline src/cloudevents.rs:33]: This is failing our tests (because symbol is unused)

**Review by AnotherDaniel — COMMENTED** [review]
One question, about copying the cloudevents.proto definition into this repo.

File: src/cloudevents.rs (hunk 1 above)
Comment (sophokles73) [reply src/cloudevents.rs:35]: already on it

**Comment by sophokles73** [comment]
@AnotherDaniel would you mind taking another look?

File: proto/io/cloudevents/v1/cloudevents.proto
Comment (sophokles73) [reply proto/io/cloudevents/v1/cloudevents.proto:1]: Good idea, will do.

**Comment by sophokles73** [comment]
@AnotherDaniel I have updated the NOTICE.md file accordingly.

**Review by AnotherDaniel — APPROVED** [review]
LGTM

**Pull request merged**